Sources/NIOExtras/LengthFieldBasedFrameDecoder.swift

```diff
@@ -0,0 +1,113 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+///
+/// An enumeration to describe the length of a piece of data in bytes.
+/// It is contained to lengths that can be converted to integer types.
+///
+public enum ByteLength: Int {
+    case one = 1
+    case two = 2
+    case four = 4
+    case eight = 8
```

Comment:
I don't think this `enum` needs a raw value.

Reply:
I just moved it to a fileprivate method as it is likely to be used more than once if the class evolves.